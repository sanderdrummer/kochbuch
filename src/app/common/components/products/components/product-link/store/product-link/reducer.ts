export const initialState = {
    id: '',
    productId: '',
    amount: '0',
    isActive: false
};

export type ProductLink = typeof initialState;
