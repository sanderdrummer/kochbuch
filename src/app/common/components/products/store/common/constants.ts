const NAMESPACE = '[PRODUCTS]';
export const ADD = NAMESPACE + 'ADD';
export const UPDATE = NAMESPACE + 'UPDATE';