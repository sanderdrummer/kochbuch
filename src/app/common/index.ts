
export { Shadows } from './styles';

export { SearchForm } from './components/forms';
export { LayoutSection, Card } from './components/layout';

export { ProductLinkLayout } from './components/products';

export { createUiReducer, UiState, uiActions, UiActions } from './store/ui/createUiWithNameSpace';
export { createCollectionReducer, 
  CollectionState,
  collectionActions, CollectionActions, types }  from './store/collection/createCollectionReducer';

export { Api } from './store/api/api.service';

export { AppState } from './store/state';
