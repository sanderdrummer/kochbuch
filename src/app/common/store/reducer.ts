import { combineReducers, Reducer } from 'redux';
import { AppState } from './state';
import recipes from '../../recipes/store/reducer';
import productModule from '../components/products';

const reducer: Reducer<AppState> = combineReducers({
    recipes,
    products: productModule.reducer
});

export default reducer;