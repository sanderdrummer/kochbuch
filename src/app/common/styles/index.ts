export { buttonStyle, inputStyle, hoverButton } from './forms/button';
export { Colors } from './colors';
export { Gardients } from './gardients';
export { Shadows } from './shadows';
export { linkStyle } from './links';