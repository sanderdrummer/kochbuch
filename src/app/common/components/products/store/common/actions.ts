import * as types from './constants';
import { Product } from '../product/reducer';
import { Action } from '../../../../store/action';

export type ProductAction = Action<Product>;
export const add = (payload: Product) => ({ type: types.ADD, payload });
export const update = (payload: Product) => ({ type: types.UPDATE, payload });

export type Actions = ProductAction;