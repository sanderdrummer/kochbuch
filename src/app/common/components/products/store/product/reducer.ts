import { Reducer } from 'redux';
import { Actions } from '../common/actions';
import * as types from '../common/constants';

export interface Product {
    readonly title: string;
    readonly popularity: number;
}

export const initialState = {
    title: 'New Product',
    popularity: 0
};

const reducer: Reducer<Product> = (state: Product = initialState, action: Actions) => {
    switch (action.type) {
      case types.UPDATE:
        return updateProduct(state, action.payload);

      default:
        return state;
    }
};

function updateProduct(state: Product, product: Product) {
    return {...state, ...product};
}

export default reducer;