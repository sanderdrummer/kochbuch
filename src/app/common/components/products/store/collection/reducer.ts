import { Product } from '../product/reducer';
import { Actions } from '../common/actions';
import product from '../product/reducer';
import * as types from '../common/constants';

export type Collection = {[title: string]: Product};

const reducer = (state: Collection, action: Actions) => {
    switch (action.type) {
        case types.ADD: return addProduct(state, action.payload);
        case types.UPDATE: return product(state[action.payload.title], action);
        default: return state;
    }
};

function addProduct (state: Collection, newProduct: Product) {
    return {...state, [newProduct.title]: newProduct};
}

export default reducer;