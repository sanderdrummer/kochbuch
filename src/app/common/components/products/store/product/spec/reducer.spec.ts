import reducer, { initialState } from '../reducer';
import * as actions from '../../common/actions';
import * as mocks from './mocks';

describe('Products reducer tests', () => {
    it('should update for ON_UPDATE action', () => {
        expect(reducer(initialState, actions.update(mocks.getUpdatedProduct())))
        .toMatchSnapshot();
    });
});