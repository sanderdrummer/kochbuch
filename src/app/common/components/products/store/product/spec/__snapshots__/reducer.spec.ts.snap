// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products reducer tests should update for ON_UPDATE action 1`] = `
Object {
  "popularity": 0,
  "title": "test product",
}
`;
